File: php/src/Model/Skill.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A skill in the workspace catalogue.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class Skill
{
    public function __construct(
        public readonly string $id,
        public readonly string $key,
        public readonly string $name,
        public readonly ?string $description,
        /** ISO-8601 */
        public readonly string $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            key: (string) ($data['key'] ?? ''),
            name: (string) ($data['name'] ?? ''),
            description: isset($data['description']) ? (string) $data['description'] : null,
            createdAt: (string) ($data['createdAt'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/Skill.java
package io.fivexer.sdk.model;

/**
 * A skill in the workspace catalogue.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class Skill {
    private String id;
    private String key;
    private String name;
    private String description;
    private String createdAt;  // ISO-8601

    public String getId() { return id; }
    public String getKey() { return key; }
    public String getName() { return name; }
    public String getDescription() { return description; }
    public String getCreatedAt() { return createdAt; }
}
